### PR TITLE
[Stream] Stream requestmanager fix

### PR DESCRIPTION
**Pull request by JackyWYX**
## Issue

Fix some request manager concerns and bug (https://github.com/harmony-one/harmony/issues/3589).

1. requestmanager's streams is updated with refresh instead of follow events. So that the streams in request manager is not corrupted.
2. Requests will no longer stay in waitings queue once the request is being canceled. So that there shall not be error message with 

```
{"level":"debug","module":"request manager","error":"no more available streams","request":"REQUEST [GetBlocksByHashes: d1e57d5ddaaa816670c8aa9c0e4ed12c490e515af2a4915c4100af238a205ddc]","caller":"/home/yx/go/src/github.com/harmony-one/harmony/p2p/stream/common/requestmanager/requestmanager.go:293","time":"2021-03-16T04:39:32.377309174Z","message":"No available streams."}
```

## Tests

Unit tests added for each scenario.

### CODE REVIEW

**Review by rlan35 — APPROVED** [review]

File: p2p/stream/common/requestmanager/requestmanager.go [hunk 1]
```diff
@@ -349,26 +348,49 @@ func (rm *requestManager) pickAvailableStream(req *request) (*stream, error) {
...
 	rm.lock.Lock()
 	defer rm.lock.Unlock()
 
-	if _, ok := rm.streams[st.ID()]; !ok {
-		rm.streams[st.ID()] = &stream{Stream: st}
-		rm.available[st.ID()] = struct{}{}
+	added, removed := checkStreamUpdates(rm.streams, rm.sm.GetStreams())
+
+	for _, st := range added {
+		rm.logger.Info().Str("streamID", string(st.ID())).Msg("add new stream")
```
Comment (LeoHChen) [inline p2p/stream/common/requestmanager/requestmanager.go:358]: may change to debug level , IMO.

File: p2p/stream/common/requestmanager/requestmanager.go [hunk 2]
```diff
@@ -349,26 +348,49 @@ func (rm *requestManager) pickAvailableStream(req *request) (*stream, error) {
...
+		rm.addNewStream(st)
+	}
+	for _, st := range removed {
+		rm.logger.Info().Str("streamID", string(st.ID())).Msg("remove stream")
```
Comment (LeoHChen) [inline p2p/stream/common/requestmanager/requestmanager.go:362]: may change to debug level , IMO.

File: p2p/stream/common/requestmanager/requestmanager.go (hunk 1 above)
Comment (JackyWYX) [reply p2p/stream/common/requestmanager/requestmanager.go:358]: These log messages may appear a lot in an unstable network (which is the case of error injected in stn) where peer connection is unstable thus a lot of connect/disconnect operation is executed. For stable network, this shall not appear a lot. Please let me know if this log appears a lot in a stable network.

File: p2p/stream/common/requestmanager/requestmanager.go (hunk 2 above)
Comment (JackyWYX) [reply p2p/stream/common/requestmanager/requestmanager.go:362]: These log messages may appear a lot in an unstable network (which is the case of error injected in stn) where peer connection is unstable thus a lot of connect/disconnect operation is executed. For stable network, this shall not appear a lot. Please let me know if this log appears a lot in a stable network.

**Pull request merged**